apps/segment/src/pages/api/webhooks/order-cancelled.ts

```diff
@@ -0,0 +1,66 @@
+import { createSegmentClientForWebhookContext } from "@/modules/create-segment-client-for-webhook-context";
+import { trackingEventFactory } from "@/modules/tracking-events/tracking-events";
+import { saleorApp } from "@/saleor-app";
+import { NextWebhookApiHandler, SaleorAsyncWebhook } from "@saleor/app-sdk/handlers/next";
+import {
+  OrderCancelledDocument,
+  OrderUpdatedSubscriptionPayloadFragment,
+} from "../../../../generated/graphql";
+import * as Sentry from "@sentry/nextjs";
+import { SegmentNotConfiguredError } from "@/errors";
+import { createLogger } from "@saleor/apps-shared";
+
+export const config = {
+  api: {
+    bodyParser: false,
+  },
+};
+
+export const orderCancelledWebhook =
+  new SaleorAsyncWebhook<OrderUpdatedSubscriptionPayloadFragment>({
+    name: "Order Cancelled v1",
+    webhookPath: "api/webhooks/order-cancelled",
+    event: "ORDER_CANCELLED",
+    apl: saleorApp.apl,
+    query: OrderCancelledDocument,
+  });
+
+const logger = createLogger({ name: "orderCancelledWebhook" });
+
+const handler: NextWebhookApiHandler<OrderUpdatedSubscriptionPayloadFragment> = async (
+  req,
+  res,
+  context,
+) => {
+  const { authData, payload } = context;
+
+  if (!payload.order) {
+    Sentry.captureException(new Error("Order not found in payload"));
+
+    return res.status(400).end();
```

Comment:
**Issue[^1][^2]**:
Status 400 will cause event retry, which will never succeed.

[^1]: 🛑 Blocking the review. 
[^2]: Possible bug or not considered application state.

Reply:
Let's get rid of these type of problems for good and create shared `WebhookResponse` utility.

Reply:
@peelar I'm not sure if we will find a common abstraction. I don't see a reason to have utility that will abstract `rest.status.send`

@krzysztofwolski right, its a problem. However this should never happen. I will return 200 and Sentry should inform us. `order_cancelled` should always have this field available.

Reply:
Updated to send 200